plugins/module/create_service_account.py

```diff
@@ -0,0 +1,152 @@
+                }
+message:
+    description: The output error / exception message that is returned in the case the module generates an error / exception.
+    type: dict
+    returned: in case of error / exception
+srvce_acc_resp_obj: 
+    description: The service account response object
+    type: dict
+    returned: when service account is created successfully
+    sample: Client ID and Client Secret of the service account. 
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import rhoas_service_accounts_mgmt_sdk
+import auth.rhoas_auth as auth
+from rhoas_service_accounts_mgmt_sdk.api import service_accounts_api
+from rhoas_service_accounts_mgmt_sdk.model.service_account_create_request_data import ServiceAccountCreateRequestData
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        name=dict(type='str', required=True),
+        description=dict(type='str', required=True),
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # changed is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        original_message='',
+        message='',
+        srvce_acc_resp_obj=dict
+        
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    # module.check_mode = False
+    if module.check_mode:
+        module.exit_json(**result)
+
+    token = auth.get_access_token()
+    
+    configuration = rhoas_service_accounts_mgmt_sdk.Configuration(
+        host = "https://sso.redhat.com/auth/realms/redhat-external",
+    )
+    configuration.access_token = token["access_token"]
+
+    
+    # Enter a context with an instance of the API client
+    with rhoas_service_accounts_mgmt_sdk.ApiClient(configuration) as api_client:
+        # Create an instance of the API class
+        api_instance = service_accounts_api.ServiceAccountsApi(api_client)
+        try:
+            service_account_create_request_data = ServiceAccountCreateRequestData(
+                name=module.params['name'],
+                description=module.params['description'],
+            ) 
+            api_response = api_instance.create_service_account(service_account_create_request_data)
+            
+            print('type api_response', api_response)
```

Comment:
We should not print those values!!

Reply:
Specifically  api_response['secret'] is `secret` and should not be in output log

Reply:
See https://www.redhat.com/sysadmin/ansible-playbooks-secrets

Reply:
you're right. damn must have left it in from debugging.

Reply:
@wtrocki so regardless, the output of the playbook run is returned and printed to screen, to use the vars `client_id` and `client_secret` we must return them. 
The alternative as per the Ansible docs is to run the play book with the `no_log` param set as per https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#how-do-i-keep-secret-data-in-my-playbook. 
this censors the output but allows the value to be returned. I'm after looking but I really do not see another way of doing this

Reply:
I don't like it but I can't find another way to do it